### PR TITLE
Implement missing SymUnmanaged* interfaces

### LINKED ISSUE #66637: Cannot use BenchmarkDotNet with NativeAOT
### Description

I'm using BenchmarkDotNet with NativeAOT, but it failed to build and yielded the error message:
```
  ...
  ILC: Method '[System.Management]System.Management.WbemObjectTextSrc..ctor()' will always throw because: Invalid IL or CLR metadata in 'Void System.Management.WbemObjectTextSrc..ctor()'
  ILC: Method '[System.Management]System.Management.WbemContext..ctor()' will always throw because: Invalid IL or CLR metadata in 'Void System.Management.WbemContext..ctor()'
  ILC: Method '[System.Management]System.Management.WbemLocator..ctor()' will always throw because: Invalid IL or CLR metadata in 'Void System.Management.WbemLocator..ctor()'
  ILC: Method '[System.Management]System.Management.WbemDefPath..ctor()' will always throw because: Invalid IL or CLR metadata in 'Void System.Management.WbemDefPath..ctor()'
  ILC: Method '[System.Management]System.Management.WbemStatusCodeText..ctor()' will always throw because: Invalid IL or CLR metadata in 'Void System.Management.WbemStatusCodeText..ctor()'
  <unknown>:0: error: assembler label '' can not be undefined
  <unknown>:0: error: assembler label '' can not be undefined
  ...
  <unknown>:0: error: assembler label '' can not be undefined
  <unknown>:0: error: assembler label '' can not be undefined
  <unknown>:0: error: assembler label '' can not be undefined
  <unknown>:0: error: assembler label '' can not be undefined
  <unknown>:0: error: assembler label '' can not be undefined
  <unknown>:0: error: assembler label '' can not be undefined
  <unknown>:0: error: assembler label '' can not be undefined
  <unknown>:0: error: assembler label '' can not be undefined
  <unknown>:0: error: assembler label '' can not be undefined
  <unknown>:0: error: assembler label '' can not be undefined
EXEC : error : The method or operation is not implemented. [...\bin\Release\net6.0\bf30ca63-8eeb-4379-8a70-8cf7fe931ae7\BenchmarkDotNet.Autogenerated.csproj]
  System.NotImplementedException: The method or operation is not implemented.
     at Internal.TypeSystem.Ecma.UnmanagedPdbSymbolReader.SymUnmanagedReaderWrapperCache.SymUnmanagedReaderRcw.GetMethod(Int32 methodToken, ISymUnmanagedMethod& method)
     at Internal.TypeSystem.Ecma.UnmanagedPdbSymbolReader.GetSequencePointsForMethod(Int32 methodToken)+MoveNext()
     at ILCompiler.DependencyAnalysis.MethodCodeNode.GetNativeSequencePoints()+MoveNext()
     at ILCompiler.DependencyAnalysis.ObjectWriter.BuildDebugLocInfoMap(ObjectNode node)
     at ILCompiler.DependencyAnalysis.ObjectWriter.EmitObject(String objectFilePath, IReadOnlyCollection`1 nodes, NodeFactory factory, ObjectWritingOptions options, IObjectDumper dumper, Logger logger)
     at ILCompiler.RyuJitCompilation.CompileInternal(String outputFile, ObjectDumper dumper)
     at ILCompiler.Compilation.ILCompiler.ICompilation.Compile(String outputFile, ObjectDumper dumper)
     at ILCompiler.Program.Run(String[] args)
     at ILCompiler.Program.Main(String[] args)
```

### Reproduction Steps

```csharp
var config = DefaultConfig.Instance
    .AddJob(Job.ShortRun
        .WithToolchain(CoreRtToolchain.CreateBuilder()
            .UseCoreRtNuGet(
                microsoftDotNetILCompilerVersion: "7.0.0-*",
                nuGetFeedUrl: "https://pkgs.dev.azure.com/dnceng/public/_packaging/dotnet7/nuget/v3/index.json")
            .DisplayName("NativeAOT")
            .TargetFrameworkMoniker("net6.0")
            .ToToolchain()));
BenchmarkRunner.Run<Test>(config);

public class Test
{
    [Benchmark]
    public void SomeBenchmarkCode() { ... }
}
```

Then run it with `dotnet run -c Release`.

### Expected behavior

Successfully run benchmark.

### Actual behavior

Failed to build benchmark

### Regression?

_No response_

### Known Workarounds

_No response_

### Configuration

_No response_

### Other information

_No response_

### PR DESCRIPTION
**Pull request by hez2010**
This PR implements missing RCW COMWrappers for SymUnmanaged* interfaces required by compiling BenchmarkDotNet with NativeAOT.

Fixes #66637 

/cc: @agocke

### CODE REVIEW

File: src/coreclr/tools/Common/TypeSystem/Ecma/SymbolReader/UnmanagedPdbSymbolReader.cs [hunk 1]
```diff
@@ -364,10 +364,782 @@ private void DisposeInternal()
...
+            public unsafe sealed record SymUnmanagedNamespaceRcw(IntPtr Inst) : ISymUnmanagedNamespace
+            {
+                private bool _disposed = false;
+
+                public int GetName(int bufferLength, out int count, char[] name)
+                {
+                    var func = (delegate* unmanaged<IntPtr, int, int*, char*, int>)(*(*(void***)Inst + 3));
+                    int hr;
+
+                    fixed (int* countPtr = &count)
```
Comment (hez2010) [inline src/coreclr/tools/Common/TypeSystem/Ecma/SymbolReader/UnmanagedPdbSymbolReader.cs:392]: Should I pin the locals? Or can I just pass the `out count` as is and use `delegate* unmanaged<IntPtr, int, out int, char*, int` for such cases?

File: src/coreclr/tools/Common/TypeSystem/Ecma/SymbolReader/UnmanagedPdbSymbolReader.cs [hunk 2]
```diff
@@ -364,10 +364,782 @@ private void DisposeInternal()
...
+                    {
+                        return;
+                    }
+                    _disposed = true;
+                    Marshal.Release(Inst);
+                }
+
+                ~SymUnmanagedNamespaceRcw()
+                {
+                    DisposeInternal();
```
Comment (hez2010) [inline src/coreclr/tools/Common/TypeSystem/Ecma/SymbolReader/UnmanagedPdbSymbolReader.cs:492]: Didn't find out a proper way to dispose as need, so I leave the clean-up work to GC.

File: src/coreclr/tools/Common/TypeSystem/Ecma/SymbolReader/UnmanagedPdbSymbolReader.cs [hunk 3]
```diff
@@ -364,10 +364,782 @@ private void DisposeInternal()
...
+                    var func = (delegate* unmanaged<IntPtr, ISymUnmanagedDocument, int, int, int*, int>)(*(*(void***)Inst + 7));
+                    fixed (int* offsetPtr = &offset)
+                    {
+                        return func(Inst, document, line, column, offsetPtr);
+                    }
+                }
+
+                public int GetRanges(ISymUnmanagedDocument document, int line, int column, int bufferLength, out int count, int[] ranges)
+                {
+                    var func = (delegate* unmanaged<IntPtr, ISymUnmanagedDocument, int, int, int, int*, int*, int>)(*(*(void***)Inst + 8));
```
Comment (hez2010) [inline src/coreclr/tools/Common/TypeSystem/Ecma/SymbolReader/UnmanagedPdbSymbolReader.cs:999]: Oops I missed this one. Will change to use `GCHandle` for `ISymUnmanagedDocument` marshalling.

Comment (hez2010) [inline src/coreclr/tools/Common/TypeSystem/Ecma/SymbolReader/UnmanagedPdbSymbolReader.cs:999]: Oops I missed this one. Will change to use `GCHandle` for `ISymUnmanagedDocument` marshalling.

File: src/coreclr/tools/Common/TypeSystem/Ecma/SymbolReader/UnmanagedPdbSymbolReader.cs [hunk 4]
```diff
@@ -364,10 +365,799 @@ private void DisposeInternal()
...
+                private bool _disposed = false;
+
+                public int GetName(int bufferLength, out int count, char[] name)
+                {
+                    var func = (delegate* unmanaged<IntPtr, int, int*, char*, int>)(*(*(void***)Inst + 3));
+                    int hr;
+
+                    fixed (int* countPtr = &count)
+                    {
+                        if (name == null)
```
Comment (jkotas) [inline src/coreclr/tools/Common/TypeSystem/Ecma/SymbolReader/UnmanagedPdbSymbolReader.cs:395]: This check is unnecessary. Pinning of null `char[]` produces null pointer.

(Multiple places in this PR. I am not commenting on all of them.)

File: src/coreclr/tools/Common/TypeSystem/Ecma/SymbolReader/UnmanagedPdbSymbolReader.cs [hunk 5]
```diff
@@ -364,10 +365,799 @@ private void DisposeInternal()
...
+                {
+                    var func = (delegate* unmanaged<IntPtr, Guid*, int>)(*(*(void***)Inst + methodSolt));
+                    int hr;
+
+                    fixed (Guid* guidPtr = &guid)
+                    {
+                        hr = func(Inst, guidPtr);
+                    }
+
+                    return hr;
```
Comment (jkotas) [inline src/coreclr/tools/Common/TypeSystem/Ecma/SymbolReader/UnmanagedPdbSymbolReader.cs:831]:
```suggestion
                    fixed (Guid* guidPtr = &guid)
                    {
                        return func(Inst, guidPtr);
                    }
```
to match the style of other wrappers.

File: src/coreclr/tools/Common/TypeSystem/Ecma/SymbolReader/UnmanagedPdbSymbolReader.cs (hunk 4 above)
Comment (hez2010) [reply src/coreclr/tools/Common/TypeSystem/Ecma/SymbolReader/UnmanagedPdbSymbolReader.cs]: Done.

**Review by agocke — APPROVED** [review]
LGTM, thanks!

**Comment by agocke** [comment]
@jkotas Anything else you wanted to look at?

**Review by jkotas — APPROVED** [review]
LGTM

**Pull request merged**